<launch>
  <!-- =========================
       Load robot description
       ========================= -->
  <param name="robot_description" command="$(find xacro)/xacro '$(find arm100_description)/urdf/so100.urdf'" />

  <!-- =========================
       Load controller configs (parameters only, no spawning)
       ========================= -->
  <rosparam file="$(find arm100_hardware)/config/controllers.yaml" command="load" />

  <!-- =========================
       Start hardware interface node only
       ========================= -->
  <node name="so_arm_100_hw_node"
        pkg="arm100_hardware"
        type="so_arm_100_hw_node"
        output="screen">
    <param name="serial_port" value="/dev/ttyACM1" />
    <param name="serial_baudrate" value="1000000" />
    <param name="use_sim_time" value="false" />
    <param name="calibration_file" value="$(find arm100_moveit_config)/config/joint_offsets.yaml" />
  </node>
</launch>
